pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -718,6 +721,74 @@ const (
 	minSafeProxySize = 2
 )
 
+func (cr *PerconaXtraDBCluster) setProbesDefaults() {
+	if cr.Spec.PXC != nil {
+		if cr.Spec.PXC.LivenessInitialDelaySeconds != nil {
+			cr.Spec.PXC.LivenessProbes.InitialDelaySeconds = *cr.Spec.PXC.LivenessInitialDelaySeconds
+		} else if cr.Spec.PXC.LivenessProbes.InitialDelaySeconds == 0 {
+			cr.Spec.PXC.LivenessProbes.InitialDelaySeconds = 300
+		}
```

Comment:
just Q: why it is needed to convert 0 to 300?

Reply:
E.g. we do not have `spec.pxc.livenessProbe.initialDelaySeconds` option at all in CR we have '0' here but need to change it to our custom default value (300). That is why we have such checks.